src/wh_server_keystore.c

```
@@ -1100,6 +1216,132 @@ _HandleUnwrapAndCacheKeyRequest(whServerContext*                         server,
     /* Cache the key */
     return wh_Server_KeystoreCacheKey(server, &metadata, key);
 }
+
+static int _HandleDataWrapRequest(whServerContext*                   server,
+                                  whMessageKeystore_DataWrapRequest* req,
+                                  uint8_t* reqData, uint32_t reqDataSz,
+                                  whMessageKeystore_DataWrapResponse* resp,
+                                  uint8_t* respData, uint32_t respDataSz)
+{
+
+    int      ret;
+    uint8_t* wrappedData;
+    uint8_t  data[WOLFHSM_CFG_KEYWRAP_MAX_DATA_SIZE];
+
+    if (server == NULL || req == NULL || reqData == NULL || resp == NULL ||
+        respData == NULL || req->dataSz > WOLFHSM_CFG_KEYWRAP_MAX_DATA_SIZE) {
+        return WH_ERROR_BADARGS;
+    }
+
+    /* Check if the reqData is big enough to hold the data */
+    if (reqDataSz < req->dataSz) {
+        return WH_ERROR_BUFFER_SIZE;
+    }
+
+    /* Extract the metadata and data from reqData */
+    memcpy(data, reqData, req->dataSz);
+
+    /* Store the wrapped data in the response data */
+    wrappedData = respData;
+
+    switch (req->cipherType) {
+
+#ifndef NO_AES
+#ifdef HAVE_AESGCM
+        case WC_CIPHER_AES_GCM: {
+            uint16_t wrappedDataSz = WOLFHSM_KEYWRAP_AES_GCM_IV_SIZE +
+                                     WOLFHSM_KEYWRAP_AES_GCM_TAG_SIZE +
+                                     req->dataSz;
+
+            /* Check if the response data can fit the wrapped data */
+            if (respDataSz < wrappedDataSz) {
```

we don't want to return this from this function (essentially crashing the server if anything bad happens) but instead need to propagate the failure back to the client. Look at the changes I made to this PR to the regular keywrap functions below (which I did to get it to pass CI)